plugins/woocommerce-blocks/assets/js/blocks-registry/product-collection/register-product-collection.tsx

```diff
@@ -26,17 +27,359 @@ export interface ProductCollectionConfig extends BlockVariation {
 	};
 }
 
+const isValidCollectionConfig = ( config: ProductCollectionConfig ) => {
+	// Basic checks for the top-level argument
+	if ( typeof config !== 'object' || config === null ) {
+		console.warn(
+			'Invalid arguments: You must pass an object to __experimentalRegisterProductCollection.'
+		);
+	}
+
+	/**
+	 * BlockVariation properties validation
+	 */
+	// name
+	if ( typeof config.name !== 'string' || config.name.length === 0 ) {
+		console.warn( 'Invalid name: name must be a non-empty string.' );
```

Comment:
As above, I think we should use error and bail early when the property is required

```suggestion
		console.error( 'Invalid name: name must be a non-empty string.' );
		return false;
```